### PR TITLE
[extractor/yappy] Support /profile/ URLs 

### PR DESCRIPTION
**Pull request by 7vlad7**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

[extractor/yappy] Added support for /profile

Fixes #

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
🐶📹🔄

Add support for extracting videos from Yappy profile pages. Create a new extractor class `YappyProfileIE` in `yappy.py` and import it in `_extractors.py`.

> _To scrape Yappy profiles with ease_
> _We import a new class from `yappy.py`_
> _It handles multiple videos per page_
> _With `itertools` we can iterate_
> _And we fix a small indent mistake_

### Walkthrough
*  Add a new extractor for Yappy profile pages ([link](https://github.com/yt-dlp/yt-dlp/pull/7346/files?diff=unified&w=0#diff-780b22dc7eb280f5a7b2bbf79aff17826de88ddcbf2fc1116ba19901827aa4e3L2429-R2432), [link](https://github.com/yt-dlp/yt-dlp/pull/7346/files?diff=unified&w=0#diff-a5b01da69730894bf1fb17132f616056ccf0200c666dffe4be2c6329156da881R8), [link](https://github.com/yt-dlp/yt-dlp/pull/7346/files?diff=unified&w=0#diff-a5b01da69730894bf1fb17132f616056ccf0200c666dffe4be2c6329156da881R101-R136))
* Fix indentation error in `YappyIE` class ([link](https://github.com/yt-dlp/yt-dlp/pull/7346/files?diff=unified&w=0#diff-a5b01da69730894bf1fb17132f616056ccf0200c666dffe4be2c6329156da881L54-R57))

</details>

### CODE REVIEW

File: yt_dlp/extractor/yappy.py [hunk 1]
```diff
@@ -51,9 +52,9 @@ def _real_extract(self, url):
         nextjs_data = self._search_nextjs_data(webpage, video_id)
 
         media_data = (
-            traverse_obj(
-                nextjs_data, ('props', 'pageProps', ('data', 'OpenGraphParameters')), get_all=False)
-            or self._download_json(f'https://yappy.media/api/video/{video_id}', video_id))
+                traverse_obj(
+                    nextjs_data, ('props', 'pageProps', ('data', 'OpenGraphParameters')), get_all=False)
+                or self._download_json(f'https://yappy.media/api/video/{video_id}', video_id))
```
Comment (bashonly) [inline yt_dlp/extractor/yappy.py:57]: revert indentation, this is what the linter was complaining about

**Comment by turnme0n** [comment]
ураа люпимый брад давай ты сможешь

File: yt_dlp/extractor/yappy.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
     unified_timestamp,
     url_or_none
 )
+import itertools
```
Comment (bashonly) [inline yt_dlp/extractor/yappy.py:8]: move to top of file

File: yt_dlp/extractor/yappy.py [hunk 3]
```diff
@@ -97,3 +98,39 @@ def _real_extract(self, url):
...
+            entries.extend([
+                self.url_result(
+                    f'https://yappy.media/video/{video["uuid"]}',
+                    ie=YappyIE.ie_key(),
+                    video_id=video['uuid'],
+                    video_title=video.get('description') or video['uuid'])
+                for video in profile_data['results']
+            ])
+
+        return self.playlist_result(entries, profile_id)
```
Comment (bashonly) [inline yt_dlp/extractor/yappy.py:136]: It's better to use a generator for this.
You can use `traverse_obj` to safely traverse the JSON object and filter for dicts with a `uuid` key
```suggestion
        def entries(profile_id):
            for page_num in itertools.count(1):
                videos = traverse_obj(self._download_json(
                    f'https://yappy.media/api/video/list/{profile_id}?page={page_num}',
                    profile_id, f'Downloading profile page {page_num} JSON'),
                    ('results', lambda _, v: v['uuid']))
                if not videos:
                    break
                for video in videos:
                    yield self.url_result(
                        f'https://yappy.media/video/{video["uuid"]}', YappyIE,
                        video['uuid'], video.get('description'))

        return self.playlist_result(entries(profile_id), profile_id)
```

**Review by bashonly — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/_extractors.py [hunk 4]
```diff
@@ -2426,7 +2426,10 @@
     ZenYandexChannelIE,
 )
 from .yapfiles import YapFilesIE
-from .yappy import YappyIE
+from .yappy import (
+    YappyIE,
+    YappyProfileIE
```
Comment (bashonly) [inline yt_dlp/extractor/_extractors.py:2431]:
```suggestion
    YappyProfileIE,
```

**Review by bashonly — APPROVED** [review]

File: yt_dlp/extractor/yappy.py [hunk 5]
```diff
@@ -97,3 +99,32 @@ def _real_extract(self, url):
...
+                videos = traverse_obj(self._download_json(
+                    f'https://yappy.media/api/video/list/{profile_id}?page={page_num}',
+                    profile_id, f'Downloading profile page {page_num} JSON'),
+                    ('results', lambda _, v: v['uuid']))
+                if not videos:
+                    break
+                for video in videos:
+                    yield self.url_result(
+                        f'https://yappy.media/video/{video["uuid"]}', YappyIE,
+                        video['uuid'], video.get('description'))
```
Comment (pukkandan) [inline yt_dlp/extractor/yappy.py:128]: Does each page have a diff number of videos? If so, can merge. If the number is fixed, use pagedlist.

Comment (7vlad7) [reply yt_dlp/extractor/yappy.py:128]: > Does each page have a diff number of videos? If so, @bashonly can merge. If the number is fixed, use pagedlist.

All pages (except the last one) contain 15 videos, the last page may have less than 15

Comment (pukkandan) [reply yt_dlp/extractor/yappy.py:128]: Then we can use `PagedList`. This allows users to download videos from the middle `-I 30-40` without extra requests.

**Review by bashonly — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/yappy.py [hunk 6]
```diff
@@ -3,7 +3,8 @@
     int_or_none,
     traverse_obj,
     unified_timestamp,
-    url_or_none
+    url_or_none,
+    OnDemandPagedList,
```
Comment (bashonly) [inline yt_dlp/extractor/yappy.py:7]: move `OnDemandPagedList` to the top of utils imports
sort classes before functions, then sort alphabetically (or just use `isort`)

**Review by bashonly — APPROVED** [review]

**Pull request merged**